File: RootTools2/src/main/java/com/rarnu/tools/neo/xposed/ads/FuckCalendar.java
package com.rarnu.tools.neo.xposed.ads;

import android.content.Context;
import android.content.Intent;
import com.rarnu.tools.neo.xposed.XpUtils;
import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.callbacks.XC_LoadPackage;
import org.json.JSONObject;

/**
 * Created by rarnu on 11/18/16.
 */
public class FuckCalendar {

    public static void fuckCalendar(XC_LoadPackage.LoadPackageParam loadPackageParam) {
        XpUtils.findAndHookMethod("com.miui.calendar.ad.AdUtils", loadPackageParam.classLoader, "canShowBrandAd", Context.class, XC_MethodReplacement.returnConstant(false));
        XpUtils.findAndHookMethod("com.miui.calendar.ad.AdService", loadPackageParam.classLoader, "onHandleIntent", Intent.class, XC_MethodReplacement.returnConstant(null));
        XpUtils.findAndHookMethod("com.miui.calendar.card.single.local.SummarySingleCard", loadPackageParam.classLoader, "needShowAdBanner", XC_MethodReplacement.returnConstant(false));
        XpUtils.findAndHookMethod("com.miui.calendar.card.single.custom.ad.AdSingleCard", loadPackageParam.classLoader, "needDisplay", XC_MethodReplacement.returnConstant(false));
        XpUtils.findAndHookMethod("com.miui.calendar.ad.AdUtils", loadPackageParam.classLoader, "getAdConfigJson", Context.class, XC_MethodReplacement.returnConstant(null));
        XpUtils.findAndHookMethod("com.miui.calendar.card.single.custom.RecommendSingleCard", loadPackageParam.classLoader, "needDisplay", XC_MethodReplacement.returnConstant(false));
        XpUtils.findAndHookMethod("com.miui.calendar.card.single.custom.ad.LargeImageAdSingleCard", loadPackageParam.classLoader, "needDisplay", XC_MethodReplacement.returnConstant(false));
        Class<?> clsC = XpUtils.findClass(loadPackageParam.classLoader, "com.xiaomi.ad.internal.common.module.a$c");
        if (clsC != null) {
            XpUtils.findAndHookMethod("com.xiaomi.ad.internal.common.module.a", loadPackageParam.classLoader, "b", clsC, XC_MethodReplacement.returnConstant(null));
        }
        XpUtils.findAndHookMethod("com.xiaomi.ad.common.pojo.Ad", loadPackageParam.classLoader, "parseJson", JSONObject.class, XC_MethodReplacement.returnConstant(null));
        XpUtils.findAndHookMethod("com.xiaomi.ad.internal.a.e", loadPackageParam.classLoader, "onAdInfo", String.class, XC_MethodReplacement.returnConstant(null));
        XpUtils.findAndHookMethod("com.miui.calendar.util.DiskStringCache", loadPackageParam.classLoader, "getString", Context.class, String.class, new XC_MethodHook() {
            @Override
            protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
                String key = (String) param.args[1];

                if (key.startsWith("bottom_banner_is_closed_today")) {
                    param.setResult("true");
                }
            }
            @Override
            protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                String key = (String) param.args[1];
                if (key.startsWith("bottom_banner_is_closed_today")) {
                    param.setResult("true");
                }
            }
        });
    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/xposed/ads/FuckCalendar.kt
package com.rarnu.tools.neo.xposed.ads

import android.content.Context
import android.content.Intent
import com.rarnu.xfunc.*
import org.json.JSONObject

/**
 * Created by rarnu on 11/18/16.
 */
object FuckCalendar {

    fun fuckCalendar(pkg: XposedPkg) {
        pkg.findClass("com.miui.calendar.ad.AdUtils").apply {
            findMethod("canShowBrandAd", Context::class.java).hook { replace { result = false } }
            findMethod("getActionBarBitmap", Context::class.java).hook { replace { result = null } }
            findMethod("getAdConfigJson", Context::class.java).hook { replace { result = null } }
        }
        pkg.findClass("com.miui.calendar.ad.AdService").findMethod("onHandleIntent", Intent::class.java).hook { replace { result = null } }
        pkg.findClass("com.miui.calendar.card.single.local.SummarySingleCard").findMethod("needShowAdBanner").hook { replace { result = false } }
        pkg.findClass("com.miui.calendar.card.single.custom.ad.AdSingleCard").findMethod("needDisplay").hook { replace { result = false } }
        pkg.findClass("com.miui.calendar.card.single.custom.HistorySingleCard").findMethod("needDisplay").hook { replace { result = false } }
        pkg.findClass("com.miui.calendar.huangli.HuangLiDetailActivity").findMethod("startLoading").hook { replace { result = null } }
        pkg.findClass("com.miui.calendar.card.single.custom.RecommendSingleCard").findMethod("needDisplay").hook { replace { result = false } }
        pkg.findClass("com.miui.calendar.card.single.custom.ad.LargeImageAdSingleCard").findMethod("needDisplay").hook { replace { result = false } }

        val clsC = pkg.findClass("com.xiaomi.ad.internal.common.module.a\$c")
        if (clsC != null) {
            pkg.findClass("com.xiaomi.ad.internal.common.module.a").findMethod("b", clsC).hook { replace { result = null } }
        }
        pkg.findClass("com.xiaomi.ad.common.pojo.Ad").findMethod("parseJson", JSONObject::class.java).hook { replace { result = null } }
        pkg.findClass("com.xiaomi.ad.internal.a.e").findMethod("onAdInfo", String::class.java).hook { replace { result = null } }
        pkg.findClass("com.miui.calendar.util.DiskStringCache").findMethod("getString", Context::class.java, String::class.java).hook {
            before {
                val key = args[1] as String
                if (key.startsWith("bottom_banner_is_closed_today")) {
                    result = "true"
                }
            }
            after {
                val key = args[1] as String
                if (key.startsWith("bottom_banner_is_closed_today")) {
                    result = "true"
                }
            }
        }
    }
}
